### PR TITLE
feat(gatsby): Add structured error to warn windows users about uppercase folders

### PR DESCRIPTION
**Pull request by djD-REK**
## Description

- Catches a GraphQL error that happens on Windows when there are uppercase folders in the path
- Warns the user that the error could be caused by uppercase folders
- Implemented as an error #85902 based on error #85901

## Background

- I found this when trying to install codesandbox on Windows 7 using yarn
- My directory structure was C:/GitHub/djD-REK and I got the following error
image
- Renaming my directory structure to C:/github/djd-rek fixed the error
image

## Rationale

- I added this warning message because it was very confusing to try to fix this error.
- After Googling, I found the following issue, which taught me how to fix it.
- I thought a warning message would be more useful.

## Related Issues

image

### CODE REVIEW

**Comment by djD-REK** [comment]
== Updated commit to add win-node-env
Further investigation of my bug installing codesandbox-client (a gatsby site) using gatsby revealed that the lowercase directories are necessary, but not sufficient -- I still need the "win-node-env" package to avoid the RelayParser error about duplicate errors. So Windows users using both win-node-env and a lowercase directory structure can install and run gatsby sites.

== Above test failing?
Also I'm not sure why that test failed, it does not seem related to my code changes

File: packages/gatsby/src/query/error-parser.js [hunk 1]
```diff
@@ -24,6 +24,22 @@ const errorParser = ({ message, filePath, location }) => {
...
+    // each document must have a unique name. Duplicated documents:"
+    {
+      regex: /Error: RelayParser: Encountered duplicated.*/gm,
+      cb: match => {
+        return {
+          id: `85902`,
+          context: {
+            sourceMessage:
+              match[0] +
+              `\n\nWarning: Windows users may be able to fix this error by renaming all directories to lowercase\nin the file path and also by installing package node-win-env using npm install -g node-win-env`,
```
Comment (wardpeet) [inline packages/gatsby/src/query/error-parser.js]: I'm not sure what `node-win-env` would fix here? `node-win-env` fixes env-variables syntax. I can't find in their readme that it fixes this error.

**Comment by LekoArts** [comment]
Hi @djD-REK 	👋 	A friendly nudge from us in case you haven't seen Ward's questions 😊

**Comment by pieh** [comment]
Hey @djD-REK, can you tell us more about your setup, I tried to reproduce but this but couldn't.

**Comment by djD-REK** [comment]
Thanks for getting back to me! I upgraded from Windows 7 to Windows 10 and
can no longer reproduce the issue. Thanks for your help! The feature
request can be closed.

Best regards,
Derek

Derek R. Austin, PT, DPT, MS, BCTMB, CSCS


Join me on LinkedIn: https://www.linkedin.com/in/derek-austin/

**Comment by sidharthachatterjee** [comment]
@djD-REK No worries! Thank you so much for your work on this nonetheless 💜 

Closing this for now

**Pull request closed without merging**